mender-auth/cli/keystore.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_AUTH_KEYSTORE_HPP
+#define MENDER_AUTH_KEYSTORE_HPP
+
+#include <common/error.hpp>
+#include <common/crypto.hpp>
+
+namespace mender {
+namespace auth {
+namespace cli {
+
+namespace error = mender::common::error;
+namespace crypto = mender::common::crypto;
+
+using namespace std;
+
+enum KeyStoreErrorCode {
+	NoError = 0,
+	NoKeysError,
+	StaticKeyError,
+};
+
+class KeyStoreErrorCategoryClass : public std::error_category {
+public:
+	const char *name() const noexcept override;
+	string message(int code) const override;
+};
+extern const KeyStoreErrorCategoryClass KeyStoreErrorCategory;
+
+error::Error MakeError(KeyStoreErrorCode code, const string &msg);
+
+enum StaticKey {
```

Comment:
Nit: `enum class` gives better type safety (int and bool are incompatible). If you change it, then you can probably change the values to `No` and `Yes` as well, because including the type name is mandatory when using `enum class`.

Reply:
+1

Reply:
Thank you. :memo: